src/socket.rs

```diff
@@ -22,18 +21,14 @@ where
 }
 
 impl AsyncRead for Socket {
-    #[inline]
-    unsafe fn prepare_uninitialized_buffer(&self, _buf: &mut [MaybeUninit<u8>]) -> bool {
-        false
-    }
-
     #[inline]
     fn poll_read(
         mut self: Pin<&mut Self>,
         cx: &mut Context<'_>,
-        buf: &mut [u8],
-    ) -> Poll<io::Result<usize>> {
-        Pin::new(&mut self.0).poll_read(cx, buf)
+        buf: &mut ReadBuf<'_>,
```

Comment:
It seems this implementation is wrong, you can refer to [this](https://github.com/Hexilee/roa/blob/master/roa/src/stream.rs#L15) to implement the newest `tokio::io::AsyncRead`.

Reply:
Done. Copied verbatim; I'm happy to not understand this.